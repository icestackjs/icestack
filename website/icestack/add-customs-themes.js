const { sharedExtraColors } = require('@icestack/ui/defaults')
const { defu } = require('@icestack/ui/shared')
/**
 * @type {import('@icestack/ui/types').Themes}
 */
const defaultThemes = {
  cupcake: {
    types: {
      primary: '#65c3c8',
      secondary: '#ef9fbc',
      accent: '#eeaf3a',
      neutral: '#291334'
    },
    extraColors: {
      ...sharedExtraColors.light,
      'base-100': '#faf7f5',
      'base-200': '#efeae6',
      'base-300': '#e7e2df',
      'base-content': '#291334'
    }
  },
  nord: {
    types: {
      primary: '#5E81AC',
      secondary: '#81A1C1',
      accent: '#88C0D0',
      neutral: '#4C566A',
      // 'neutral-content': '#D8DEE9',
      info: '#B48EAD',
      success: '#A3BE8C',
      warning: '#EBCB8B',
      error: '#BF616A'
    },
    extraColors: {
      ...sharedExtraColors.light,
      'base-100': '#ECEFF4',
      'base-200': '#E5E9F0',
      'base-300': '#D8DEE9',
      'base-content': '#2E3440'
    }
  },
  dracula: {
    types: {
      primary: '#ff79c6',
      secondary: '#bd93f9',
      accent: '#ffb86c',
      neutral: '#414558',
      info: '#8be9fd',
      success: '#50fa7b',
      warning: '#f1fa8c',
      error: '#ff5555'
    },
    extraColors: {
      ...sharedExtraColors.dark,
      'base-100': '#282a36',
      'base-content': '#f8f8f2'
    }
  },
  dim: {
    types: {
      primary: '#9FE88D',
      secondary: '#FF7D5C',
      accent: '#C792E9',
      neutral: '#1c212b',
      // 'neutral-content': '#B2CCD6',
      info: '#28ebff',
      success: '#62efbd',
      warning: '#efd057',
      error: '#ffae9b'
    },
    extraColors: {
      ...sharedExtraColors.dark,
      'base-100': '#2A303C',
      'base-200': '#242933',
      'base-300': '#20252E',
      'base-content': '#B2CCD6'
    }
  }
}

function addCustomThemes(...args) {
  return defu(...args, defaultThemes)
}
/**
 * @returns {import('@icestack/ui/types').Themes}
 */
module.exports = {
  addCustomThemes
}
