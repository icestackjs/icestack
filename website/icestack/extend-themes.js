const { defu } = require('@icestack/ui/shared')

/**
 * @type {import('@icestack/ui/types').Themes}
 */
const overrideDefaultSelector = {
  light: {
    selector: 'html,:root',
  },
  dark: {
    selector: '.dark',
  },
}

/**
 * @type {import('@icestack/ui/types').Themes}
 */
const extendColorsTypes = {
  light: {
    types: {
      // auto generate colors

      info: '#2dd4bf',
      accent: '#22c55e',
      secondary: '#2f54eb',
    },
  },
  dark: {
    types: {
      // is dark: true
      info: ['#2dd4bf', true],
      accent: [
        '#22c55e',
        {
          theme: 'dark',
          backgroundColor: '#000000',
        },
      ],
      // full control
      secondary: {
        'secondary': '#2f54eb',
        'secondary-content': '#ffffff',
        'secondary-hover': '#263ea0',
        'secondary-active': '#5273e0',
      },
    },
  },
}

function extendThemes(...args) {
  return defu(...args, extendColorsTypes, overrideDefaultSelector)
}

/**
 * @returns {import('@icestack/ui/types').Themes}
 */
module.exports = {
  extendThemes,
}
