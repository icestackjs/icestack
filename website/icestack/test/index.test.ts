import { addCustomThemes } from '../add-customs-themes'
import { extendThemes } from '../extend-themes'

describe('index', () => {
  it('addCustomThemes', () => {
    expect(addCustomThemes()).toMatchSnapshot()
  })

  it('extendThemes', () => {
    expect(extendThemes()).toMatchSnapshot()
  })

  it('compose case 0', () => {
    const obj0 = extendThemes(addCustomThemes())
    const obj1 = addCustomThemes(extendThemes())
    expect(obj0).toEqual(obj1)
    expect(obj0).toMatchSnapshot()
  })
})
