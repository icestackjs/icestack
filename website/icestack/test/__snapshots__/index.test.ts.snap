// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > addCustomThemes 1`] = `
{
  "cupcake": {
    "extraColors": {
      "base-100": "#faf7f5",
      "base-1000": "#262626",
      "base-1100": "#1f1f1f",
      "base-1200": "#141414",
      "base-1300": "#000000",
      "base-200": "#efeae6",
      "base-300": "#e7e2df",
      "base-400": "#f0f0f0",
      "base-500": "#d9d9d9",
      "base-600": "#bfbfbf",
      "base-700": "#8c8c8c",
      "base-800": "#595959",
      "base-900": "#434343",
      "base-content": "#291334",
    },
    "types": {
      "accent": "#eeaf3a",
      "neutral": "#291334",
      "primary": "#65c3c8",
      "secondary": "#ef9fbc",
    },
  },
  "dim": {
    "extraColors": {
      "base-100": "#2A303C",
      "base-1000": "#f0f0f0",
      "base-1100": "#f5f5f5",
      "base-1200": "#fafafa",
      "base-1300": "#ffffff",
      "base-200": "#242933",
      "base-300": "#20252E",
      "base-400": "#262626",
      "base-500": "#434343",
      "base-600": "#595959",
      "base-700": "#8c8c8c",
      "base-800": "#bfbfbf",
      "base-900": "#d9d9d9",
      "base-content": "#B2CCD6",
    },
    "types": {
      "accent": "#C792E9",
      "error": "#ffae9b",
      "info": "#28ebff",
      "neutral": "#1c212b",
      "neutral-content": "#B2CCD6",
      "primary": "#9FE88D",
      "secondary": "#FF7D5C",
      "success": "#62efbd",
      "warning": "#efd057",
    },
  },
  "dracula": {
    "extraColors": {
      "base-100": "#282a36",
      "base-1000": "#f0f0f0",
      "base-1100": "#f5f5f5",
      "base-1200": "#fafafa",
      "base-1300": "#ffffff",
      "base-200": "#141414",
      "base-300": "#1f1f1f",
      "base-400": "#262626",
      "base-500": "#434343",
      "base-600": "#595959",
      "base-700": "#8c8c8c",
      "base-800": "#bfbfbf",
      "base-900": "#d9d9d9",
      "base-content": "#f8f8f2",
    },
    "types": {
      "accent": "#ffb86c",
      "error": "#ff5555",
      "info": "#8be9fd",
      "neutral": "#414558",
      "primary": "#ff79c6",
      "secondary": "#bd93f9",
      "success": "#50fa7b",
      "warning": "#f1fa8c",
    },
  },
  "nord": {
    "extraColors": {
      "base-100": "#ECEFF4",
      "base-1000": "#262626",
      "base-1100": "#1f1f1f",
      "base-1200": "#141414",
      "base-1300": "#000000",
      "base-200": "#E5E9F0",
      "base-300": "#D8DEE9",
      "base-400": "#f0f0f0",
      "base-500": "#d9d9d9",
      "base-600": "#bfbfbf",
      "base-700": "#8c8c8c",
      "base-800": "#595959",
      "base-900": "#434343",
      "base-content": "#2E3440",
    },
    "types": {
      "accent": "#88C0D0",
      "error": "#BF616A",
      "info": "#B48EAD",
      "neutral": "#4C566A",
      "neutral-content": "#D8DEE9",
      "primary": "#5E81AC",
      "secondary": "#81A1C1",
      "success": "#A3BE8C",
      "warning": "#EBCB8B",
    },
  },
}
`;

exports[`index > compose case 0 1`] = `
{
  "cupcake": {
    "extraColors": {
      "base-100": "#faf7f5",
      "base-1000": "#262626",
      "base-1100": "#1f1f1f",
      "base-1200": "#141414",
      "base-1300": "#000000",
      "base-200": "#efeae6",
      "base-300": "#e7e2df",
      "base-400": "#f0f0f0",
      "base-500": "#d9d9d9",
      "base-600": "#bfbfbf",
      "base-700": "#8c8c8c",
      "base-800": "#595959",
      "base-900": "#434343",
      "base-content": "#291334",
    },
    "types": {
      "accent": "#eeaf3a",
      "neutral": "#291334",
      "primary": "#65c3c8",
      "secondary": "#ef9fbc",
    },
  },
  "dark": {
    "selector": ".dark",
    "types": {
      "accent": [
        "#22c55e",
        {
          "backgroundColor": "#000000",
          "theme": "dark",
        },
      ],
      "info": [
        "#2dd4bf",
        true,
      ],
      "secondary": {
        "secondary": "#2f54eb",
        "secondary-active": "#5273e0",
        "secondary-content": "#ffffff",
        "secondary-hover": "#263ea0",
      },
    },
  },
  "dim": {
    "extraColors": {
      "base-100": "#2A303C",
      "base-1000": "#f0f0f0",
      "base-1100": "#f5f5f5",
      "base-1200": "#fafafa",
      "base-1300": "#ffffff",
      "base-200": "#242933",
      "base-300": "#20252E",
      "base-400": "#262626",
      "base-500": "#434343",
      "base-600": "#595959",
      "base-700": "#8c8c8c",
      "base-800": "#bfbfbf",
      "base-900": "#d9d9d9",
      "base-content": "#B2CCD6",
    },
    "types": {
      "accent": "#C792E9",
      "error": "#ffae9b",
      "info": "#28ebff",
      "neutral": "#1c212b",
      "neutral-content": "#B2CCD6",
      "primary": "#9FE88D",
      "secondary": "#FF7D5C",
      "success": "#62efbd",
      "warning": "#efd057",
    },
  },
  "dracula": {
    "extraColors": {
      "base-100": "#282a36",
      "base-1000": "#f0f0f0",
      "base-1100": "#f5f5f5",
      "base-1200": "#fafafa",
      "base-1300": "#ffffff",
      "base-200": "#141414",
      "base-300": "#1f1f1f",
      "base-400": "#262626",
      "base-500": "#434343",
      "base-600": "#595959",
      "base-700": "#8c8c8c",
      "base-800": "#bfbfbf",
      "base-900": "#d9d9d9",
      "base-content": "#f8f8f2",
    },
    "types": {
      "accent": "#ffb86c",
      "error": "#ff5555",
      "info": "#8be9fd",
      "neutral": "#414558",
      "primary": "#ff79c6",
      "secondary": "#bd93f9",
      "success": "#50fa7b",
      "warning": "#f1fa8c",
    },
  },
  "light": {
    "selector": "html,:root",
    "types": {
      "accent": "#22c55e",
      "info": "#2dd4bf",
      "secondary": "#2f54eb",
    },
  },
  "nord": {
    "extraColors": {
      "base-100": "#ECEFF4",
      "base-1000": "#262626",
      "base-1100": "#1f1f1f",
      "base-1200": "#141414",
      "base-1300": "#000000",
      "base-200": "#E5E9F0",
      "base-300": "#D8DEE9",
      "base-400": "#f0f0f0",
      "base-500": "#d9d9d9",
      "base-600": "#bfbfbf",
      "base-700": "#8c8c8c",
      "base-800": "#595959",
      "base-900": "#434343",
      "base-content": "#2E3440",
    },
    "types": {
      "accent": "#88C0D0",
      "error": "#BF616A",
      "info": "#B48EAD",
      "neutral": "#4C566A",
      "neutral-content": "#D8DEE9",
      "primary": "#5E81AC",
      "secondary": "#81A1C1",
      "success": "#A3BE8C",
      "warning": "#EBCB8B",
    },
  },
}
`;

exports[`index > extendThemes 1`] = `
{
  "dark": {
    "selector": ".dark",
    "types": {
      "accent": [
        "#22c55e",
        {
          "backgroundColor": "#000000",
          "theme": "dark",
        },
      ],
      "info": [
        "#2dd4bf",
        true,
      ],
      "secondary": {
        "secondary": "#2f54eb",
        "secondary-active": "#5273e0",
        "secondary-content": "#ffffff",
        "secondary-hover": "#263ea0",
      },
    },
  },
  "light": {
    "selector": "html,:root",
    "types": {
      "accent": "#22c55e",
      "info": "#2dd4bf",
      "secondary": "#2f54eb",
    },
  },
}
`;
